package/samplers/moead/_child_generation_strategy.py

```diff
@@ -0,0 +1,178 @@
+from optuna.samplers.nsgaii._crossover import _try_crossover
+from optuna.samplers.nsgaii._crossovers._base import BaseCrossover
+from optuna.trial import FrozenTrial
+
+
+_NUMERICAL_DISTRIBUTIONS = (
+    FloatDistribution,
+    IntDistribution,
+)
+
+
+class MOEAdChildGenerationStrategy:
+    """Generate a child parameter from the given parent population by MOEA/D algorithm.
+    Args:
+        study:
+            Target study object.
+        search_space:
+            A dictionary containing the parameter names and parameter's distributions.
+        parent_population:
+            A list of trials that are selected as parent population.
+    Returns:
+        A dictionary containing the parameter names and parameter's values.
+    """
+
+    def __init__(
+        self,
+        *,
+        mutation_prob: float | None = None,
+        crossover: BaseCrossover,
+        crossover_prob: float,
+        swapping_prob: float,
+        rng: LazyRandomState,
+    ) -> None:
+        if not (mutation_prob is None or 0.0 <= mutation_prob <= 1.0):
+            raise ValueError(
+                "`mutation_prob` must be None or a float value within the range [0.0, 1.0]."
+            )
+
+        if not (0.0 <= crossover_prob <= 1.0):
+            raise ValueError("`crossover_prob` must be a float value within the range [0.0, 1.0].")
+
+        if not (0.0 <= swapping_prob <= 1.0):
+            raise ValueError("`swapping_prob` must be a float value within the range [0.0, 1.0].")
+
+        if not isinstance(crossover, BaseCrossover):
+            raise ValueError(
+                f"'{crossover}' is not a valid crossover."
+                " For valid crossovers see"
+                " https://optuna.readthedocs.io/en/stable/reference/samplers.html."
+            )
+
+        self._crossover_prob = crossover_prob
+        self._mutation_prob = mutation_prob
+        self._swapping_prob = swapping_prob
+        self._crossover = crossover
+        self._rng = rng
+        self._subproblem_id = 0
+
+    def __call__(
+        self,
+        study: Study,
+        search_space: dict[str, BaseDistribution],
+        parent_population: list[FrozenTrial],
+        neighbors: dict[int, list[int]],
+    ) -> dict[str, Any]:
+        """Generate a child parameter from the given parent population by NSGA-II algorithm.
+        Args:
+            study:
+                Target study object.
+            search_space:
+                A dictionary containing the parameter names and parameter's distributions.
+            parent_population:
+                A list of trials that are selected as parent population.
+            neighbors:
+                A dictionary containing the subproblem id and its neighboring subproblems.
+        Returns:
+            A dictionary containing the parameter names and parameter's values.
+        """
+        subproblem_parent_population = [
+            parent_population[i] for i in neighbors[self._subproblem_id]
+        ]
```

Comment:
I'm not sure, but this implementation might have unexpected behavior in the case of multi-threading.